R/rf_modeling.R

```diff
@@ -175,26 +176,46 @@ rf_modeling <- function( msnset, features, response, pred.cls, K=NULL, sel.feat=
     sel.alg <- match.arg(sel.alg)
     FUN <- switch(sel.alg,
                   varSelRF = select_features_varSelRF,
-                  Boruta = function(x,y) select_features_Boruta(x,y,...),
+                  Boruta = function(x,y, ...) select_features_Boruta(x,y,...),
                   top = select_features_top)
 
     # do K-fold split here
     if(is.null(K))
         K <- nrow(dSet)
     num_rep <- ceiling(nrow(dSet)/K)
+    set.seed(seed)
     cv_idx <- sample(rep(seq_len(K), num_rep)[seq_len(nrow(dSet))])
-    multiproc_cl <- makeCluster(max(1, detectCores() - 1))
-    clusterEvalQ(multiproc_cl, library("MSnID"))
-    clusterEvalQ(multiproc_cl, library("Biobase"))
-    silence <- clusterExport(multiproc_cl,
+    if(is.null(cores)){
+        cores <- max(1, detectCores() - 1)
+    }
+    stopifnot(1 <= cores)
+    if(cores > detectCores()){
+        msg <- sub("\n", "", "The number of specified processes is greater than
+                the number of cores available on this computer. 
+                This may lead to high computational overhead.")
+        warning(msg)
+    }
+    multiproc_cl <- makeCluster(cores, outfile = "")
+
+    clusterEvalQ(multiproc_cl, invisible(suppressWarnings({
+        Sys.setenv(`_R_S3_METHOD_REGISTRATION_NOTE_OVERWRITES_` = "false")
+        suppressWarnings(suppressPackageStartupMessages({
+            library("MSnID")
+            library("Biobase")
+        }))
+    })))
+    invisible(clusterExport(multiproc_cl,
                              c("dSet","cv_idx","features",
                                "response"),
-                             envir = environment())
+                             envir = environment()))
     fn <- function(i){
+       RNGkind("L'Ecuyer-CMRG")
+       set.seed(i)
+       seed <- i
```

Comment:
Set random number generator kind and random seed in worker function by its index (`i`).